CONTRIBUTING.md

```diff
@@ -16,7 +16,110 @@ For the Kubeflow Trainer documentation, please check [the official Kubeflow docu
 
 Note for Lima the link is to the Adopters, which supports several different container environments.
 
-## Development
+## Contributing Workflow
+For contributions to the Kubeflow Trainer project, please follow this workflow:
+
+1. Fork the repository on GitHub
+2. Clone your fork locally
+3. Add the upstream remote to keep your fork in sync:
+   ```sh
+   git remote add upstream https://github.com/kubeflow/trainer.git
+   ```
+4. Create a new branch for your changes:
+   ```sh
+   git checkout -b feature/your-feature-name
+   ```
+5. Fetch the latest changes from the upstream repository and rebase your branch:
+   ```sh
+   git fetch upstream
+   git rebase upstream/main
+   ```
+6. Make your changes and commit them:
+   - Install pre-commit hooks before creating commits:
+     ```sh
+     pip install pre-commit
+     pre-commit install
+     ```
+     ```sh
+     git commit -m -s "your commit message"
+     ```
+   - The pre-commit hooks will automatically check your code for style issues
+   - If the hooks fail, fix the issues (if they weren't fixed automatically) and stage your changes again
+   - For more details, see the [Code Style](#code-style) section
+7. Push your changes to your fork and create a pull request (PR)
+
+
+# Building the Operator
+Install dependencies:
+
+```sh
+go mod tidy
+```
+
+Build the library:
+
+```sh
+go install github.com/kubeflow/trainer/cmd/trainer-controller-manager
+```
+
+## Running the Operator Locally
+
+Running the operator locally in a docker container (as opposed to deploying it on a K8s cluster) is convenient for debugging/development.
+
+### Run a Kubernetes cluster
```

Comment:
We don't really need it, since this instruction creates Kind cluster for you: https://github.com/kubeflow/trainer/blob/c31bdbdf684b223df0e66a5e80b288ed50f8879a/CONTRIBUTING.md#e2e-tests